application/templates/deployment.yaml

```
@@ -276,10 +276,12 @@ spec:
         {{- end }}
         {{- end }}
       {{- end }}
+      {{- if .Values.rbac }}
+      {{- if .Values.rbac.serviceAccount }}
       {{- if .Values.rbac.serviceAccount.name }}
```

I think we can leverage order of evaluation here to simplify this line. This seems to be working as of Helm v3.11.1:

```suggestion
      {{- if and .Values.rbac .Values.rbac.serviceAccount .Values.rbac.serviceAccount.name }}
```

However I think we should instead just make sure `.Values.rbac.serviceAccount` is at least `{}` in our `values.yaml`, **which we already do!** So I recommend going simply with:

```suggestion
      {{- with .Values.rbac.serviceAccount.name }}
```

And using `.` within. This is because to break the above line, one would have to provide overrides like:

```yaml
rbac:
```

or

```yaml
rbac:
  serviceAccount:
```

At that point, I'd expect the chart user to know what they're doing and understand what to do when template rendering fails.

---

We need to define default service account `{{ template "application.name" $ }}` in else block incase `.Values.rbac.serviceAccount.name` isnt defined, using `if` is fine

---

What do you think about going forward as below

```suggestion
      {{- if .Values.rbac.serviceAccount.enabled }}
      {{- if $.Values.rbac.serviceAccount.name }}
      serviceAccountName: {{ $.Values.rbac.serviceAccount.name }}
      {{- else }}
      serviceAccountName: {{ template "application.name" $ }}
      {{- end }}
      {{- end }}
```